libr/bin/format/mach0/mach0.c

```diff
@@ -1744,6 +1744,162 @@ static int inSymtab(SdbHt *hash, struct symbol_t *symbols, const char *name, ut6
 	return false;
 }
 
+void MACH0_(pull_symbols)(struct MACH0_(obj_t)* mo, RBinSymbolCallback cb, void *user) {
+	const char *symstr;
+	int j, s, stridx, symbols_size, symbols_count;
+	ut32 to, from, i;
+	//ut64 text_base = get_text_base (bin);
+
+	if (!mo || !mo->symtab || !mo->symstr) {
+		return;
+	}
+	/* parse symbol table */
+	/* parse dynamic symbol table */
+	symbols_count = (mo->dysymtab.nextdefsym + \
+			mo->dysymtab.nlocalsym + \
+			mo->dysymtab.nundefsym );
+	symbols_count += mo->nsymtab;
+	//symbols_count = mo->nsymtab;
+	symbols_size = (symbols_count + 1) * 2 * sizeof (struct symbol_t);
+
+	if (symbols_size < 1) {
+		return;
+	}
+	SdbHt *hash = sdb_ht_new ();
+	for (s = j = 0; s < 2; s++) {
+		switch (s) {
+		case 0:
+			from = mo->dysymtab.iextdefsym;
+			to = from + mo->dysymtab.nextdefsym;
+			break;
+		case 1:
+			from = mo->dysymtab.ilocalsym;
+			to = from + mo->dysymtab.nlocalsym;
+			break;
+#if NOT_USED
+		case 2:
+			from = mo->dysymtab.iundefsym;
+			to = from + mo->dysymtab.nundefsym;
+			break;
+#endif
+		}
+		if (from == to) {
+			continue;
+		}
+		from = R_MIN (R_MAX (0, from), symbols_size / sizeof (struct symbol_t));
```

Comment:
`symbols_size ...`  can be stored once maybe

Reply:
?

Reply:
meant `size_t something = symbols_size / sizeof (struct symbol_t)`